crates/flycheck/src/lib.rs

```diff
@@ -533,7 +535,7 @@ impl CargoActor {
 }
 
 enum CargoMessage {
-    CompilerArtifact(cargo_metadata::Artifact),
+    CompilerArtifact(Box<cargo_metadata::Artifact>),
```

Comment:
Last time this came up (can't find the PR, though), it was pointed out that this is the most common variant, so boxing the contents will cause almost every instance to incur an allocation.

I don't know if this overhead will be larger than the speed-up obtained from the faster moves. Either way, I suspect it doesn't matter too much compared to the whole `cargo check` invocation.

Reply:
Ah, I see. I will keep it in my mind. Thank you for your review.

Reply:
@lnicola I have removed Box in
[dd84306](https://github.com/rust-lang/rust-analyzer/pull/15615/commits/dd843060f9ebff3862dc48c033c3ddbabb3b6a84) . Thanks